packages/bns/docs/modules.md

```diff
@@ -0,0 +1,385 @@
+
+___
+
+### registerName
+
+▸ **registerName**(`__namedParameters`: [*RegisterNameOptions*](interfaces/registernameoptions.md)): *Promise*<[*Result*](modules.md#result)\>
+
+Generates and broadcasts a name registration transaction.
+Second and final step in registering a name.
+
+Returns a Result object which will indicate if the transaction was successfully broadcast
+
+#### Parameters:
+
+Name | Type |
+:------ | :------ |
+`__namedParameters` | [*RegisterNameOptions*](interfaces/registernameoptions.md) |
+
+**Returns:** *Promise*<[*Result*](modules.md#result)\>
+
+Defined in: [index.ts:557](https://github.com/blockstack/stacks.js/blob/master/packages/bns/src/index.ts#L557)
+
+___
+
+### renewName
+
+▸ **renewName**(`__namedParameters`: [*RenewNameOptions*](interfaces/renewnameoptions.md)): *Promise*<[*Result*](modules.md#result)\>
+
+Generates and broadcasts a name renew transaction.
+This renews a name registration.
+
+Returns a Result object which will indicate if the transaction was successfully broadcasted
+
+#### Parameters:
+
+Name | Type |
+:------ | :------ |
+`__namedParameters` | [*RenewNameOptions*](interfaces/renewnameoptions.md) |
+
+**Returns:** *Promise*<[*Result*](modules.md#result)\>
+
+Defined in: [index.ts:778](https://github.com/blockstack/stacks.js/blob/master/packages/bns/src/index.ts#L778)
+
+___
+
+### revealNamespace
+
+▸ **revealNamespace**(`__namedParameters`: [*RevealNamespaceOptions*](interfaces/revealnamespaceoptions.md)): *Promise*<[*Result*](modules.md#result)\>
+
+Generates and broadcasts a namespace reveal transaction.
+Second step in registering a namespace.
+
+Returns a Result object which will indicate if the transaction was successfully broadcasted.
+
+#### Parameters:
+
+Name | Type |
+:------ | :------ |
+`__namedParameters` | [*RevealNamespaceOptions*](interfaces/revealnamespaceoptions.md) |
+
+**Returns:** *Promise*<[*Result*](modules.md#result)\>
+
+Defined in: [index.ts:331](https://github.com/blockstack/stacks.js/blob/master/packages/bns/src/index.ts#L331)
+
+___
+
+### revokeName
+
+▸ **revokeName**(`__namedParameters`: [*RevokeNameOptions*](interfaces/revokenameoptions.md)): *Promise*<[*Result*](modules.md#result)\>
+
+Generates and broadcasts a name revoke transaction.
+This revokes a name registration.
+
+Returns a Result object which will indicate if the transaction was successfully broadcasted
+
+#### Parameters:
+
+Name | Type |
+:------ | :------ |
+`__namedParameters` | [*RevokeNameOptions*](interfaces/revokenameoptions.md) |
+
+**Returns:** *Promise*<[*Result*](modules.md#result)\>
+
+Defined in: [index.ts:725](https://github.com/blockstack/stacks.js/blob/master/packages/bns/src/index.ts#L725)
+
+___
+
+### transferName
+
+▸ **transferName**(`__namedParameters`: [*TransferNameOptions*](interfaces/transfernameoptions.md)): *Promise*<[*Result*](modules.md#result)\>
+
+Generates and broadcasts a name transfer transaction.
+This changes the owner of the registered name.
```

Comment:
Should we add a note, stipulating that the underlying NFT will be transferred, meaning a post condition is required?